arch/x86/pagetables.c

```diff
@@ -147,6 +164,10 @@ static mfn_t get_pgentry_mfn(mfn_t tab_mfn, pt_index_t index, unsigned long flag
         tab = init_map_mfn(mfn);
         memset(tab, 0, PAGE_SIZE);
     }
+    else {
```

Comment:
I would move the unlikely() from the inner function and turn it into likely on the if above.